web/src/features/time/TimeController.tsx

```diff
@@ -63,6 +64,25 @@ export default function TimeController({ className }: { className?: string }) {
     }
   }, [data, datetimes, setEndDatetime, setSelectedDatetime, setStartDatetime]);
 
+  // Sync the url to the datetime returned by the backend
+  useEffect(() => {
+    if (!datetimes || !urlDatetime) {
+      return;
+    }
+
+    const endDatetime = datetimes.at(-1);
+    if (!endDatetime) {
+      return;
```

Comment:
nit: We might want to think about how to handle this case in the very unlikely chance we hit it